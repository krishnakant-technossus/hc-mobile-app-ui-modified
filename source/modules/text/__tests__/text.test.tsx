// npx jest ./source/modules/text/__tests__/text.test.ts

import * as React from 'react';
import {render} from '@testing-library/react-native';

import Text from '../text';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Text', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Text>Paragraph goes here</Text>
        </ThemeProvider>,
      );

      const text = wrapper.getByTestId('text');

      expect(text).toBeDefined();
      expect(text.children).toEqual(['Paragraph goes here']);

      expect(wrapper).toMatchSnapshot();
    });

    describe('Variants', () => {
      it('isMuted: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text isMuted>Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('isCaption: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text isCaption>Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('isSubtitle: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text isSubtitle>Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('inversed: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text inversed>Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('alignment-left: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text alignment="left">Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('alignment-center: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text alignment="center">Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('alignment-right: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Text alignment="right">Paragraph goes here</Text>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
