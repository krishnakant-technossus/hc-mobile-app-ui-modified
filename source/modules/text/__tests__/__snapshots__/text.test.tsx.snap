// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Rendering Variants alignment-center: should match snapshot 1`] = `
<Text
  alignment="center"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "center",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants alignment-left: should match snapshot 1`] = `
<Text
  alignment="left"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants alignment-right: should match snapshot 1`] = `
<Text
  alignment="right"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "right",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants inversed: should match snapshot 1`] = `
<Text
  inversed={true}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants muted: should match snapshot 1`] = `
<Text
  muted={true}
  style={
    Array [
      Object {
        "color": "#97A4AB",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants subtitle: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  subtitle={true}
  testID="text"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering Variants variant="caption": should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  testID="text"
  variant="caption"
>
  Paragraph goes here
</Text>
`;

exports[`Text Rendering renders itself correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
    ]
  }
  testID="text"
>
  Paragraph goes here
</Text>
`;
